xdsl/transforms/varith_transformations.py

```diff
@@ -64,6 +75,36 @@ def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
                 return
 
 
+class VarithToArithPattern(RewritePattern):
+    """
+    Splits a varith operation into two arith operations.
+    """
+
+    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
+        # check that the op is of a type that we can convert to arith
+        if type(op) not in VARITH_TO_ARITH_TYPE_MAP:
+            return
+
+        # this must be true, as all keys of VARITH_TO_ARITH_TYPE_MAP are varith ops
+        op = cast(
+            varith.VarithOp,
+            op,
+        )
```

Comment:
```suggestion
    def match_and_rewrite(self, op: varith.VarithOp, rewriter: PatternRewriter, /):
```
One really nice feature of the API, it can match on exactly what you want it to match on. This can be either the base class, or more specific types like `varith.add`